### DISCUSSION TITLE
Zabbix deployed on kubernetes monitoring external devices

### DISCUSSION THREAD

**Discussion by eduardocalfaia** [zabbix-community/helm-zabbix#56] [General]
Hi dears,
I have deployed zabbix on kubernetes, the web interface is accessible but how to connect zabbix-agent from the monitoring devices?
Thanks

**Comment by eduardocalfaia** [comment]
<img width="1388" alt="Screenshot 2023-12-12 at 00 57 41" src="https://github.com/zabbix-community/helm-zabbix/assets/3436692/58bf944a-138d-4b2f-9ad8-5001acb746fc">

**Reply by pavlovnicola** [reply]
Hi @eduardocalfaia 

Any luck with that? I am in the same situation. I have installed zabbix on K8s and I am trying to config the agents.
I went with passive at first, then I switched to active. Zabbix Server is configured as ClusterIP. I tried to configure the agent to connect to "zabbix-zabbix-server" but with no luck.

**Reply by aeciopires** [reply]
Hello @eduardocalfaia @pavlovnicola!

This is a same subject commeted in this topic: https://github.com/zabbix-community/helm-zabbix/discussions/27 

You need exposed the Zabbix Server and Proxy installed in the Kubernetes cluster to allow the access by Zabbix Agent installed out of k8s

**Comment by fibbs** [comment]
After almost a year, let me answer to this (sorry for not having paid attention before).

In order to allow a Zabbix agent running on any system in your network to be contacted by your Zabbix running in Kubernetes in Passive mode (passive meaning that Zabbix Server reaches out to the agent), you need to configure the `Server` configuration variable in the Zabbix agent's configuration file `/etc/zabbix/zabbix_agentd.conf` or `/etc/zabbix/zabbix_agent2.conf`. You will usually have to allow all your Kubernetes cluster's node IP addresses, as you cannot predict on which node the Zabbix Server pod is going to run.

```
##### Passive checks related

### Option: Server
#	List of comma delimited IP addresses, optionally in CIDR notation, or DNS names of Zabbix servers and Zabbix proxies.
#	Incoming connections will be accepted only from the hosts listed here.
#	If IPv6 support is enabled then '127.0.0.1', '::127.0.0.1', '::ffff:127.0.0.1' are treated equally
#	and '::/0' will allow any IPv4 or IPv6 address.
#	'0.0.0.0/0' can be used to allow any IPv4 address.
#	Example: Server=127.0.0.1,192.168.1.0/24,::1,2001:db8::/32,zabbix.example.com
#
# Mandatory: yes
# Default:
# Server=

Server=0.0.0.0/0
```

For active checks (and active proxies) to work properly, the agent (or the proxy) must be able to reach zabbix server via its port 10051. For this, you have several options in the `values.yaml` of this helm chart. Probably a `LoadBalancer` service or a `NodePort` is a good solution for you, but this depends entirely on your infrastructure. Once made the Zabbix server available from the outside of the cluster, wou have to configure Zabbix agent's `ServerActive` configuration option to point to the zabbix server.
